### DISCUSSION TITLE
Error in installing TESpy

### DISCUSSION THREAD

**Discussion by BaqirHashmi** [oemof/tespy#325] [Q&A] [answered]
I am trying to install CoolProp and Tespy libraries (Packages) in my Jupyter notebook. Firstly i was getting the following errors

"error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": [https://visualstudio.microsoft.com/visual-cpp-build-tools/](https://www.researchgate.net/deref/https%3A%2F%2Fvisualstudio.microsoft.com%2Fvisual-cpp-build-tools%2F) ERROR: Failed building wheel for CoolProp ERROR: Command errored out with exit status 1:

After that I installed Microsoft visual C++ build tools 2019 by clicking on the link given in the error prompt. Now after installing the above installer I am facing the following problems

ERROR: Command errored out with exit status 1:

c1xx: fatal error C1083: Cannot open source file: 'C:\Users\2925471\AppData\Local\Temp\pip-install-tp1rp4je\coolprop_1b4e2eea5213485b9739752d46ec1d4f\tmp5aicspg0.cpp': Permission denied

ERROR: Failed building wheel for CoolProp
[Error.docx](https://github.com/oemof/tespy/files/8319485/Error.docx)

if you could please help solve this issue , I will be really grateful

**Comment by fwitte** [comment, marked as answer]
Hi @BaqirHashmi,

welcome to the discussion forum!

Unfortunately, CoolProp is not yet available through pip or conda with Python versions >=3.9. Have you tried it with Python 3.8? Also, pip should (in theory) not allow you to install TESPy with versions of Python other than 3.7 and 3.8.

I feel there should be a clear message stating the incompatibility with specific Python versions. I‘ll try to find out, what we can do about that.

Best regards
